src/transport/raw/MessageHeader.h

```diff
@@ -35,15 +35,41 @@
 #include <lib/core/PeerId.h>
 #include <lib/support/BitFlags.h>
 #include <lib/support/BufferReader.h>
+#include <lib/support/CodeUtils.h>
 #include <lib/support/TypeTraits.h>
 #include <protocols/Protocols.h>
 #include <system/SystemPacketBuffer.h>
 
 namespace chip {
 
+namespace detail {
+// Figure out the max size of a packet we can allocate, including all headers.
+static constexpr size_t kMaxIPPacketSizeBytes     = 1280;
+static constexpr size_t kMaxPacketBufferSizeBytes = System::PacketBuffer::kMaxSizeWithoutReserve;
+static constexpr size_t kMaxPacketSizeBytes       = min(kMaxIPPacketSizeBytes, kMaxPacketBufferSizeBytes);
+
+// Figure out the max size of our headers.
+// System::PacketBuffer::kDefaultHeaderReserve may or may not include space for
+// UDP headers, depending on the situation.  Make sure we always have enough
+// space for UDP headers plus Matter headers.
+static constexpr size_t kMaxUDPHeaderSizeBytes          = 48;
+static constexpr size_t kMaxPacketbufferHeaderSizeBytes = System::PacketBuffer::kDefaultHeaderReserve;
+static constexpr size_t kMaxHeaderSizeBytes =
+    max(kMaxUDPHeaderSizeBytes + CHIP_SYSTEM_HEADER_RESERVE_SIZE, kMaxPacketbufferHeaderSizeBytes);
```

Comment:
> Wondering why the max() is necessary since 'kMaxPacketbufferHeaderSizeBytes' seems to be only including the CHIP headers

Because the bit after "since" is false @pidarped

Reply:
So to be precise:
1. `CHIP_SYSTEM_HEADER_RESERVE_SIZE` only includes the Matter headers.
2. `kMaxPacketbufferHeaderSizeBytes` always includes the Matter headers and may or may not include additional transport-level headers, depending on the situation.

Reply:
There might be other issues here, by the way, in that kMaxPacketbufferHeaderSizeBytes might include some sort of non-IP transport headers too, again depending on backend.   So 1280 - maxHeaders may be smaller than we could really do, perhaps.  

Maybe what we should really be doing is:

1. Largest conceptual payload per spec: `1280 - kMaxUDPAndIPHeaderSizeBytes - CHIP_SYSTEM_HEADER_RESERVE_SIZE`.
2. Largest payload we actually support in our packetbuffers: `PacketBuffer::kMaxSize`.
3. Take the min of those two.

@pidarped Thoughts?  That might be simpler and more correct....